lms/djangoapps/onboarding/views.py

```diff
@@ -409,9 +411,15 @@ def update_account_settings(request):
     user_extended_profile = UserExtendedProfile.objects.get(user_id=request.user.id)
     partners_opt_in = request.POST.get('partners_opt_in', '')
 
+    from lms.djangoapps.onboarding.helpers import get_user_on_demand_courses, get_email_pref_first_on_demand_courses
+    on_demand_courses = get_user_on_demand_courses(request.user)
     if request.method == 'POST':
 
-        form = forms.UpdateRegModelForm(request.POST, instance=user_extended_profile)
+        form = forms.UpdateRegModelForm(
+            [(course.id, str(course.display_name)) for course in on_demand_courses],
```

Comment:
Let's keep this list comprehension in a variable and move it a separate line, then use that variable in this constructor.